lib/glare.rb

```diff
@@ -7,39 +9,49 @@
 require 'glare/dns_record'
 require 'glare/cf_dns_records'
 require 'glare/errors'
+require 'forwardable'
 
 module Glare
   class << self
+    extend Forwardable
+
+    def_delegators :new, :register, :resolve, :deregister, :proxied?, :records
+
+    def new
+      Instance.new
+    end
+  end
+
+  class Instance
+    def initialize
+      @client = Domain.new(build_client)
+    end
+
     def register(fqdn, destination, type, proxied: false, ttl: 1)
-      client = build_client
-      Domain.new(client).register(fqdn, destination, type, proxied: proxied, ttl: ttl)
+      @client.register(fqdn, destination, type, proxied: proxied, ttl: ttl)
     end
 
     def resolve(fqdn, type)
-      client = build_client
-      Domain.new(client).resolve(fqdn, type)
+      @client.resolve(fqdn, type)
     end
 
     def deregister(fqdn, type)
-      client = build_client
-      Domain.new(client).deregister(fqdn, type)
+      @client.deregister(fqdn, type)
     end
 
     def proxied?(fqdn, type)
-      client = build_client
-      Domain.new(client).proxied?(fqdn, type)
+      @client.proxied?(fqdn, type)
     end
 
     def records(fqdn, type)
-      client = build_client
-      Domain.new(client).records(fqdn, type)
+      @client.records(fqdn, type)
     end
 
     private
 
-    CF_EMAIL = 'CF_EMAIL'.freeze
-    CF_AUTH_KEY = 'CF_AUTH_KEY'.freeze
-    CF_API_TOKEN = 'CF_API_TOKEN'.freeze
+    CF_EMAIL = 'CF_EMAIL'
```

Comment:
Just curious why no more Freeze?

Reply:
Check the header of that file, mostly `rubocop` changes